ts/components/Lightbox.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * @prettier
+ */
+import React from 'react';
+
+import classNames from 'classnames';
+
+interface Props {
+  close: () => void;
+  imageURL?: string;
+  onNext?: () => void;
+  onPrevious?: () => void;
+  onSave: () => void;
+  shouldShowNextButton: boolean;
+  shouldShowPreviousButton: boolean;
+  shouldShowSaveAsButton: boolean;
+}
+
+const styles = {
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
+    backgroundColor: 'rgba(0, 0, 0, 0.9)',
+    padding: 40,
+  } as React.CSSProperties,
+  objectContainer: {
+    flexGrow: 1,
+    display: 'inline-flex',
+    justifyContent: 'center',
+  } as React.CSSProperties,
+  image: {
+    flexGrow: 1,
+    flexShrink: 0,
+    maxWidth: '100%',
+    maxHeight: '100%',
+    objectFit: 'contain',
+  } as React.CSSProperties,
+  controls: {
+    flexShrink: 0,
+    display: 'flex',
+    flexDirection: 'column',
+    marginLeft: 10,
+  } as React.CSSProperties,
+};
+
+interface IconButtonProps {
+  type: 'save' | 'close' | 'previous' | 'next';
+  onClick?: () => void;
+}
+const IconButton = ({ onClick, type }: IconButtonProps) => (
+  <a href="#" onClick={onClick} className={classNames('iconButton', type)} />
+);
+
+export class Lightbox extends React.Component<Props, {}> {
+  public static defaultProps: Partial<Props> = {
+    shouldShowNextButton: false,
+    shouldShowPreviousButton: false,
+    shouldShowSaveAsButton: false,
+  };
+
+  private containerRef: HTMLDivElement | null = null;
+
+  public componentDidMount() {
+    const useCapture = true;
+    document.addEventListener('keyup', this.onKeyUp, useCapture);
+  }
+
+  public componentWillMount() {
+    const useCapture = true;
+    document.removeEventListener('keyup', this.onKeyUp, useCapture);
+  }
+
+  public render() {
```

Comment:
For some odd reason, I liked `render()` as the last element defined in a React component. Seems like you don't like that? How do you decide where to put it?

Reply:
Agreed, I had that too in the past but I think one of the TSLint defaults now asks to put `public` methods first which I find reasonable. We can revisit that later if you’d like.